include/lbann/utils/dnn_lib/miopen/convolution.hpp

```diff
@@ -48,15 +49,7 @@ get_fwd_conv_workspace_size(TensorDescriptor const& wDesc,
                             TensorDescriptor const& yDesc,
                             El::SyncInfo<El::Device::GPU> const& si)
 {
-  size_t size;
-  auto handle_manager = internal::make_default_handle_manager(si);
-  CHECK_MIOPEN(miopenConvolutionForwardGetWorkSpaceSize(handle_manager.get(),
-                                                        wDesc,
-                                                        xDesc,
-                                                        convDesc,
-                                                        yDesc,
-                                                        &size));
-  return size;
+  return size_t{1 << 30};
```

Comment:
So at the moment this is a performance killer that may only offer a small benefit in memory savings.  I am wondering if we should make this a runtime configuration option so that we can test to see if the MIOPEN stack fixes this bug in the future without code change on our part.  So maybe we add a runtime flag (LBANN_USE_MIOPEN_DYNAMIC_WORKSPACE_CALCULATION) or something like that in a conditional guard.

Reply:
That way we can easily test this in the future.

Reply:
As I think we've discussed offline, this implementation matches the cuDNN implementation. If the device memory allocator is caching blocks (e.g., CUB), the workspace will be reused in subsequent layers anyway. If it's not caching, this will be the least of our performance concerns.